ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -5839,13 +5842,33 @@ public HiveMetaHook getHook(
       }
     };
 
+    if ("rest".equals(conf.get("metastore.type", "hms"))) {
+      return getHiveIcebergRESTCatalog(conf, hookLoader);
+    }
+
     if (conf.getBoolVar(ConfVars.METASTORE_FASTPATH)) {
       return new SessionHiveMetaStoreClient(conf, hookLoader, allowEmbedded);
```

Comment:
Just a note. `SessionHiveMetaStoreClient` has critical traits to achieve some features, e.g., temporary tables. My question: Do we want to include those capabilities when integrating with an Iceberg REST Catalog, or do we have valid reasons why we don't need to support them?
IMetaStoreClient might not be a good place to override if we need them.